fuel-block-producer/src/block_producer.rs

```diff
@@ -55,6 +47,27 @@ use tracing::debug;
 #[cfg(test)]
 mod tests;
 
+#[derive(Error, Debug)]
+pub enum Error {
```

Comment:
For `error` interfaces do we want a name of Error across every service with type being the hint to where it's from, or should we do `<ServiceName>Error`. I think it's worth considering the latter because when debugging no matter how many weird string conversions or formatting takes place atleast the source is still preserved

Reply:
Yes that's correct, I see what you mean. My response had been to the effect that this can be tricky with serialization and stuff (as that Service module) isn't always preserved, but since we already do this for naming and other things this is fine